Add support for dplyr

cosinequanon opened the pull request:
Issue #4. @onurfiliz could you take a look when you have a chance?

onurfiliz commented on R/db.data.type.PrestoConnection.R:
```diff
@@ -0,0 +1,11 @@
+
+#' @export
+db_data_type.PrestoConnection <- function(con, fields, ...) {
+  return(vapply(fields, function(field) dbDataType(con, field), character(1)))
```
nit: character(0) should also be OK.

cosinequanon replied:
If I change it to be `character(0)` I get:
```
> db_data_type.PrestoConnection(con, 1)
Error in vapply(fields, function(field) dbDataType(con, field), character(0)) :
  values must be length 0,
 but FUN(X[[1]]) result is length 1
```

I'm just going to use `sapply` for simplicity

onurfiliz commented on R/src.presto.R:
```diff
@@ -0,0 +1,62 @@
+    port=NULL,
+    parameters=NULL,
+    ...
+  ) {
```
It might be good to check that dplyr is installed and give a descriptive error message otherwise here. It will still fail in line 49 otherwise, but with a more generic message.

onurfiliz commented:
Looks great, thank you.

Please verify that the tests pass without dplyr installed, and then feel free to merge.

cosinequanon commented:
Good call on checking the tests pass without dplyr installed, turns out `skip` throws an error if you don't supply a message. I addressed your remaining comments, and squashed all the changes so I think we are good to go.
